spec/lib/amazon_ses_mailer/message_spec.rb

```diff
@@ -0,0 +1,174 @@
+RSpec.describe AmazonSesMailer::Message do
+  let(:interceptor)   { double }
+  let(:delivery_proc) { nil }
+  let(:message)       { described_class.new({}, [interceptor], delivery_proc) }
+
+  context '.ses_client' do
+    context 'when AmazonSesMailer::Message is defined' do
+      let(:ses_client) { double }
+
+      before { described_class.ses_client = ses_client }
+
+      it 'should set client for all instances as custom ses client' do
+        expect(message.ses_client.class).to eq(ses_client.class)
+      end
+    end
+
+    context 'when AmazonSesMailer::Message not defined' do
+      before { described_class.ses_client = nil }
+
+      it 'should set client for all instances as Aws::SESV2::Client' do
+        expect(message.ses_client.class).to eq(Aws::SESV2::Client)
+      end
+    end
+  end
+
+  context '.initialize' do
+    it 'should fill message info' do
+      expect(message.message).to eq({ from_email_address: ' <>',
+                                      destination:        {
+                                        to_addresses: []
+                                      },
+                                      reply_to_addresses: [],
+                                      content:            {
+                                        template: {
+                                          template_name: nil,
+                                          template_data: nil
+                                        }
+                                      } })
+    end
+
+    it 'should fill interceptors' do
+      expect(message.instance_variable_get(:@interceptors)).to eq([interceptor])
+    end
+
+    it 'should fill delivery_proc' do
+      expect(message.instance_variable_get(:@delivery_proc).class).not_to be_nil
+    end
+  end
+
+  context '#deliver' do
```

Comment:
A lot of implementation details below, this can be simplified a lot by just ensuring the proc is either called or not.

Reply:
Not addressed.